src/languages/python.ts

```diff
@@ -130,23 +139,32 @@ const nodeMatchers: Record<ScopeType, NodeMatcher> = {
     ["list", "list_comprehension"],
     (node) => LIST_ELEMENT_TYPES.includes(node.type)
   ),
-  ifStatement: hasType("if_statement"),
-  class: possiblyDecoratedDefinition("class_definition"),
-  statement: hasType(...STATEMENT_TYPES),
-  arrowFunction: hasType("lambda"),
-  functionCall: hasType("call"),
-  argumentOrParameter: delimitedMatcher(
-    (node) =>
-      (node.parent?.type === "argument_list" &&
-        ARGUMENT_TYPES.includes(node.type)) ||
-      (PARAMETER_LIST_TYPES.includes(node.parent?.type ?? "") &&
-        PARAMETER_TYPES.includes(node.type)),
-    (node) => node.type === "," || node.type === "(" || node.type === ")",
-    ", "
+  ifStatement: typeMatcher("if_statement"),
+  class: matcher(possiblyDecoratedDefinition("class_definition")),
+  statement: typeMatcher(...STATEMENT_TYPES),
+  name: cascadingMatcher(
+    matcher(getNameNode),
+    matcher((node) => (node.type === "assignment" ? getLeftNode(node) : null))
+  ),
+  functionName: notSupported,
```

Comment:
Would it be hard to implement this one?

Reply:
I doubt it would! I think I just got distracted by what this PR turned into 😅

Reply:
ha fair enough; there was a lot going on in this PR 😄